modules/javafx.controls/src/main/java/javafx/scene/control/ContextMenu.java

```diff
@@ -246,26 +243,64 @@ public String getName() {
      public void show(Node anchor, Side side, double dx, double dy) {
         if (anchor == null) return;
         if (getItems().size() == 0) return;
 
         getScene().setNodeOrientation(anchor.getEffectiveNodeOrientation());
-        // FIXME because Side is not yet in javafx.geometry, we have to convert
-        // to the old HPos/VPos API here, as Utils can not refer to Side in the
-        // charting API.
-        HPos hpos = side == Side.LEFT ? HPos.LEFT : side == Side.RIGHT ? HPos.RIGHT : HPos.CENTER;
-        VPos vpos = side == Side.TOP ? VPos.TOP : side == Side.BOTTOM ? VPos.BOTTOM : VPos.CENTER;
-
-        // translate from anchor/hpos/vpos/dx/dy into screenX/screenY
-        Point2D point = Utils.pointRelativeTo(anchor,
-                prefWidth(-1), prefHeight(-1),
-                hpos, vpos, dx, dy, true);
-        doShow(anchor, point.getX(), point.getY());
+        setAnchorLocation(getAnchorLocation(side, anchor.getEffectiveNodeOrientation()));
+        Bounds anchorBounds = anchor.localToScreen(anchor.getLayoutBounds());
+        double x = getXBySide(anchorBounds, side, anchor.getEffectiveNodeOrientation()) + dx;
+        double y = getYBySide(anchorBounds, side) + dy;
+        doShow(anchor, x, y);
     }
 
-     /**
+     private AnchorLocation getAnchorLocation(Side side, NodeOrientation orientation) {
+         if (orientation == NodeOrientation.RIGHT_TO_LEFT) {
+             switch (side) {
+                 case TOP: return AnchorLocation.CONTENT_BOTTOM_RIGHT;
+                 case RIGHT: return AnchorLocation.CONTENT_TOP_RIGHT;
+                 case BOTTOM: return AnchorLocation.CONTENT_TOP_RIGHT;
+                 case LEFT: return AnchorLocation.CONTENT_TOP_LEFT;
+             }
+         } else {
+             switch (side) {
+                 case TOP: return AnchorLocation.CONTENT_BOTTOM_LEFT;
+                 case RIGHT: return AnchorLocation.CONTENT_TOP_LEFT;
+                 case BOTTOM: return AnchorLocation.CONTENT_TOP_LEFT;
+                 case LEFT: return AnchorLocation.CONTENT_TOP_RIGHT;
+             }
+         }
+        return AnchorLocation.CONTENT_TOP_LEFT; // never reached
+     }
+
+     private double getXBySide(Bounds anchorBounds, Side side, NodeOrientation orientation) {
+         if (orientation == NodeOrientation.RIGHT_TO_LEFT) {
+             if (side == Side.RIGHT) {
+                 return anchorBounds.getMinX();
+             } else {
+                 return anchorBounds.getMaxX();
+             }
+         } else {
+             if (side == Side.RIGHT) {
+                 return anchorBounds.getMaxX();
+             } else {
+                 return anchorBounds.getMinX();
+             }
+         }
+     }
+
+     private double getYBySide(Bounds anchorBounds, Side side) {
+         if (side == Side.BOTTOM) {
+             return anchorBounds.getMaxY();
+         } else {
+             return anchorBounds.getMinY();
+         }
+     }
+
+
+
```

Comment:
Minor: there should only be one blank line here.

Reply:
Agreed. Additional lines removed.